llvm/lib/Analysis/MemoryDependenceAnalysis.cpp

```diff
@@ -577,6 +610,9 @@ MemDepResult MemoryDependenceResults::getSimplePointerDependencyFrom(
         return MemDepResult::getDef(Inst);
       if (isInvariantLoad)
         continue;
+      if (canSkipClobberingStore(SI, MemLoc, MemLocAlign, BatchAA,
+                                 getDefaultBlockScanLimit()))
```

Comment:
Hm, do you think it would make sense to pass `Limit` here instead? That way, if we already walked ~100 instructions in the main clobber walk, we're not going to walk another ~100 trying to skip.

Reply:
Agreed, I've checked statistics on llvm-test-suite (gvn.NumGVNLoad, gvn.NumPRELoad) and seems that it will not cause any changes with this more conservative limit